hunts/views.py

```diff
@@ -214,5 +216,20 @@ def post(self, request, pk):
         return JsonResponse({'data': result})
 
 
+class ActiveHuntRedirectView(RedirectView):
+    pattern_name = 'hunts:all_puzzles'
+
+    def get_redirect_url(self, *args, **kwargs):
+        try:
+            active_hunt = Hunt.objects.get(active=True)
```

Comment:
I think this is going to raise an exception when there are multiple hunts marked active (my local db is in this state for example).

What do you think about keeping track of it per-user? Maybe it's overkill but we could maybe then update it on each page load, so whatever hunt you were last looking at is your default.

Reply:
Ooh yeah that sounds good. So, if the user's never opened smallboard we can load a generic home page (probably needed for https://github.com/cardinalitypuzzles/smallboard/issues/197 anyway as @rawxfish pointed out), otherwise redirect to the last accessed hunt. To achieve this, I'm thinking we'll want to:

- add an `active_hunt = ForeignKey(Hunt, null=True)` to the `Puzzler` user model
- update `Puzzler.active_hunt` whenever a user accesses a `Hunt`
- redirect based on `Puzzler.active_hunt` or load the home page if it's null
- remove the `Hunt.active` field

Am I missing anything?

Reply:
Some related thoughts: It'd be nice if there's somewhere users can toggle between hunts if they are working on multiple hunts simultaneously.

Also, we might consider having unique URLs per hunt so you can differentiate hunts just by looking at that. E.g.: hunt/mitmh2021/... vs. hunt/crums2020/... etc.

Another use case to think about supporting in the future (which has been discussed elsewhere) is having multiple instances of the same hunt, so different teams can use smallboard for the same hunt and not affect each other or see each other's answers. Maybe this could be accomplished by introducing the notion of a Team and having puzzlers and answer submissions associated with teams. Just something to think about.

Reply:
> Some related thoughts: It'd be nice if there's somewhere users can toggle between hunts if they are working on multiple hunts simultaneously.

Maybe something like a user settings page? Filed https://github.com/cardinalitypuzzles/smallboard/issues/203

> Also, we might consider having unique URLs per hunt so you can differentiate hunts just by looking at that. E.g.: hunt/mitmh2021/... vs. hunt/crums2020/... etc.

Do you mean a site URL per hunt? In other words, maybe a separate page per hunt?

Or do you mean enforcing a unique *external* URL per hunt?

> Another use case to think about supporting in the future (which has been discussed elsewhere) is having multiple instances of the same hunt, so different teams can use smallboard for the same hunt and not affect each other or see each other's answers. Maybe this could be accomplished by introducing the notion of a Team and having puzzlers and answer submissions associated with teams. Just something to think about.

:+1: My thoughts exactly (and @rgossiaux's too, I think). See https://github.com/cardinalitypuzzles/smallboard/issues/191 for the "elsewhere" discussions :smile:

Are any changes we should make in this PR to address the different instances per Team?

Reply:
@akirabaruah those changes sound good to me, though you could probably leave Hunt.active as it is if you wanted; not really sure about the context but I'm guessing the idea was to distinguish between hunts in-progress and completed hunts which is a little different from "which hunt a user is/was working on".

Really like the unique URL per hunt, that's a lot more readable than /2/ and stuff lol